python/ray/data/tests/test_consumption.py

```diff
@@ -1414,10 +1414,9 @@ def test_read_write_local_node_ray_client(ray_start_cluster_enabled):
     # Read/write from Ray Client will result in error.
     ray.init(address)
     with pytest.raises(ValueError):
-        ds = ray.data.read_parquet("local://" + path).materialize()
-    ds = ray.data.from_pandas(df)
-    with pytest.raises(ValueError):
-        ds.write_parquet("local://" + data_path).materialize()
+        ray.data.read_parquet("local://" + path).materialize()
+    with pytest.raises(RuntimeError):
```

Comment:
why it throws `RuntimeError` now?

Reply:
This is related to adding the check for `ray._private.worker._global_node` above. I think calling the stats actor in dataset initialization introduces an earlier reason for failure.

Reply:
This looks like we need to fix it. Can you take a deeper look of this issue? We can discuss offline.

Reply:
Does it still throw `RuntimeError`, given we added the fallback?